### PR TITLE
Update osmesa.

### PR DESCRIPTION
**Pull request by emilio**
---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14584)

### CODE REVIEW

**Comment by KiChjang** [comment]
Failed build on OS-mesa?

**Comment by emilio** [comment]
Yes, makes sense, we might need a higher LLVM version (3.3+). cc @larsbergstrom

**Comment by larsbergstrom** [comment]
I don't see any reason we couldn't upgrade LLVM. I'm assuming the version we have is just transitively pulled in implicitly by one of our dependencies.

Looking at packages.debian.org, I don't see a package for 3.3, so we'd have to do at least 3.4, if I'm reading it right.

**Comment by emilio** [comment]
arrg, mac too

**Comment by emilio** [comment]
(I don't know if the llvm update in mac is deployed or not already)

**Comment by emilio** [comment]
@aneeshusa do you know if the bots have been deployed recently? They seem to claim that LLVM is still not installed:

> configure: error: LLVM 3.3.0 or newer is required for gallium

**Comment by SimonSapin** [comment]
Unassigning myself since this this is marked "do not merge", feel free to reassign once it’s ready.

**Comment by emilio** [comment]
Yeah, @aneeshusa confirmed that this wasn't rolled out yet. Let's see now.

@bors-servo try

**Comment by emilio** [comment]
Still not rolled out? :(

**Comment by larsbergstrom** [comment]
@bors-servo try

Rolled out https://github.com/servo/saltfs/pull/582

**Comment by emilio** [comment]
> configure: error: LLVM 3.3.0 or newer is required for gallium

Great

**Comment by larsbergstrom** [comment]
@bors-servo retry

(just doing this to test https://github.com/servo/saltfs/pull/638)

**Comment by jdm** [comment]
```
--- stderr
indent: Command line: unknown parameter "--version"
Package libomxil-bellagio was not found in the pkg-config search path.
Perhaps you should add the directory containing `libomxil-bellagio.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libomxil-bellagio' found
configure: error: LLVM 3.3.0 or newer is required for gallium
```

**Comment by emilio** [comment]
@bors-servo try

 * With the newer osmesa version I only get a few linker errors when using an old zlib version (when working around the configure check). Let's see if anything else unexpected turns out.

**Comment by emilio** [comment]
Linux builders seem to work fine with the newer version

**Comment by emilio** [comment]
OSX builders hopefully only need https://github.com/servo/saltfs/pull/690, but I'll need to check when that's deployed, or learn to install the dependencies as root (brew doesn't like it a lot).

**Comment by emilio** [comment]
Building zlib manually and hacking the OSMesa makefiles to point to it makes it build \o/

**Comment by jdm** [comment]
@emilio This needs to be rebased first.

**Comment by mati865** [comment]
I rebased it and it solved build issue on Arch, Fedora and openSUSE.

Why there are no compile logs from OSX?

**Comment by jdm** [comment]
We automatically remove build logs after five days. The problem is that zlib is not a recent enough version on the build machines.

**Comment by mati865** [comment]
@jdm thanks for making it clear.

**Comment by jdm** [comment]
From the homebrew installation for zlib:
```
This formula is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

For compilers to find this software you may need to set:
    LDFLAGS:  -L/usr/local/opt/zlib/lib
    CPPFLAGS: -I/usr/local/opt/zlib/include
For pkg-config to find this software you may need to set:
    PKG_CONFIG_PATH: /usr/local/opt/zlib/lib/pkgconfig
```

File: etc/ci/buildbot_steps.yml [hunk 1]
```diff
@@ -11,26 +11,26 @@ mac-rel-wpt1:
...
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release
+  - ./mach env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig build --release
   - ./mach test-wpt --release --processes 4 --total-chunks 4 --this-chunk 4 --log-raw test-wpt.log --log-errorsummary wpt-errorsummary.log --always-succeed
   - ./mach filter-intermittents wpt-errorsummary.log --log-intermittents intermittents.log --log-filteredsummary filtered-wpt-errorsummary.log --use-tracker
 
 mac-dev-unit:
   - ./mach clean-nightlies --keep 3 --force
-  - env SERVO_RUSTC_LLVM_ASSERTIONS=1 ./mach build --dev
+  - env env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig SERVO_RUSTC_LLVM_ASSERTIONS=1 ./mach build --dev
```
Comment (aneeshusa) [inline etc/ci/buildbot_steps.yml]: There are two `envs` here

File: etc/ci/buildbot_steps.yml [hunk 2]
```diff
@@ -11,26 +11,26 @@ mac-rel-wpt1:
 
 mac-rel-wpt2:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release
+  - ./mach env PKG_CONFIG_PATH=/usr/local/opt/zlib/lib/pkgconfig build --release
```
Comment (aneeshusa) [inline etc/ci/buildbot_steps.yml]: I think env needs to go before mach here, i.e. `env PKG_CONFIG_PATH=... ./mach build --release`

**Review by aneeshusa — COMMENTED** [review]
Some fixes to hopefully get this working.

Long term it may make sense to move these env vars into saltfs along with the zlib install, I'll think more about that.

**Comment by jdm** [comment]
r=me if we squash those last two commits.

**Comment by emilio** [comment]
Awesome, thanks @jdm for figuring out the zlib on osx stuff \o/

**Comment by est31** [comment]
AWESOME!!! Waiting for this since months.

**Pull request merged**